hs_file_types/models/raster.py

```diff
@@ -209,6 +211,43 @@ def get_xml(self, pretty_print=True):
         return CoreMetaData.XML_HEADER + '\n' + etree.tostring(RDF_ROOT, encoding='UTF-8',
                                                                pretty_print=pretty_print).decode()
 
+    def get_preview_data_url(self, resource, folder_path):
+        """Generate a GeoServer layer preview link."""
+
+        if resource.raccess.public is True:
+            geoserver_url = settings.HS_GEOSERVER
+            resource_id = resource.short_id
+            layer_id = '.'.join('/'.join(folder_path.split('/')[2:]).split('.')[:-1])
+
```

Comment:
consider error handling / logging

Reply:
f214a44: Added some basic error handling and logging.